docs/api-aggregation-setup.md

```diff
@@ -0,0 +1,196 @@
+install`. They are key to set, as the signed certificates will be
+bound to the exact service name that is defined below as
+`SVCCAT_SERVICE_NAME`.
+
+```
+export HELM_NAME=catalog
+export SVCCAT_NAMESPACE=catalog
+export SVCCAT_SERVICE_NAME=${HELM_NAME}-${SVCCAT_NAMESPACE}-apiserver
+```
+ 
+## Set up the certificate bundle
+
+The APIService expects a certificate bundle. We can create our own, or
+pull the one from kube core for reuse.
+
+The certificate bundle is made up of Certificate Authority, a Serving
+Certificate, and the Serving Private Key.
+
+### Create our own new CA and generate keys
+
+This is an example. It is written with zero understanding of the best
+practices of secure certificate generation.
+
+```
+export CA_NAME=ca
+
+export ALT_NAMES="\"${SVCCAT_SERVICE_NAME}.${SVCCAT_NAMESPACE}\",\"${SVCCAT_SERVICE_NAME}.${SVCCAT_NAMESPACE}.svc"\"
+
+export SVCCAT_CA_SETUP=svc-cat-ca.json
+cat >> ${SVCCAT_CA_SETUP} << EOF
+{
+    "hosts": [ ${ALT_NAMES} ],
+    "key": {
+        "algo": "rsa",
+        "size": 2048
+    },
+    "names": [
+        {
+            "C": "US",
+            "L": "san jose",
+            "O": "kube",
+            "OU": "WWW",
+            "ST": "California"
+        }
+    ]
+}
+EOF
+
+
+cfssl genkey --initca ${SVCCAT_CA_SETUP} | cfssljson -bare ${CA_NAME}
+# now the files 'ca.csr  ca-key.pem  ca.pem' exist
+
+export SVCCAT_CA_CERT=${CA_NAME}.pem
+export SVCCAT_CA_KEY=${CA_NAME}-key.pem
+
+export PURPOSE=server
+echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+
+echo '{"CN":"'${SVCCAT_SERVICE_NAME}'","hosts":['${ALT_NAMES}'],"key":{"algo":"rsa","size":2048}}' \
+ | cfssl gencert -ca=${SVCCAT_CA_CERT} -ca-key=${SVCCAT_CA_KEY} -config=server-ca-config.json - \
+ | cfssljson -bare apiserver
+
+export SC_SERVING_CA=${SVCCAT_CA_CERT}
+export SC_SERVING_CERT=apiserver.pem
+export SC_SERVING_KEY=apiserver-key.pem
+```
+
+## Get the appropriate tls ca/cert/key from kube
+
+If you are in a cloud provider environment, you most likely do not
+have access to the appropriate keys.
+
+The key we are looking for in an already running system is the
+`--root-ca-file` flag to the controller-manager.
+
+The following is an example based on a real kubernetes cluster. The
+various files may be named differently and in different locations.
+
+
+```
+export SERVING_NAME=server-ca
+export SERVINGCA_CERT=${SERVING_NAME}.crt
+export SERVINGCA_KEY=${SERVING_NAME}.key
+# a default location
+cp /var/run/kubernetes/${SERVINGCA_CERT} .
+cp /var/run/kubernetes/${SERVINGCA_KEY} .
+```
+
+### Create a cfssl config file for a new signing key
+
+```
+export PURPOSE=server
+echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+```
+
+### Use the existing keys plus the config file to generate the new signing certificate and key
+
```

Comment:
I would note here that you **must** install the catalog into the right namespace for the certificate to be signed correctly for the service's DNS name.